mountpoint-s3/src/fs.rs

```diff
@@ -127,7 +124,15 @@ impl<Client: ObjectClient> UploadState<Client> {
                 error!("write failed: {e}");
                 // Abort the request.
                 let ret: libc::c_int = e.into();
-                *self = Self::Failed(ret);
+                match std::mem::replace(self, Self::Failed(ret)) {
+                    UploadState::InProgress { handle, .. } => {
+                        if let Err(err) = handle.finish_writing() {
+                            // Log the issue but still return the write error.
+                            error!(?err, "error updating the inode status");
+                        }
+                    }
+                    Self::Failed(_) | Self::Completed => unreachable!("checked by get_upload_in_progress"),
```

Comment:
Yes. This is equivalent to `_`, but stating the cases explicitly can make it clearer to the reader and ensures that if we ever add a new case, we need to review this `match` and decide how to handle it, instead of just falling back to the `_` arm.